src/navigation/index.jsx

```diff
@@ -83,8 +85,26 @@ const Navigation = () => {
     setupNotifications();
   }, []);
 
+  const handlingStateChange = async (state) => {
+    if (trace.current) {
+      trace.current.putMetric("endTime", Date.now()); // Record the end time before stopping (example)
+      await trace.current.stop();
+      trace.current = null;
+    }
+
+    const currentRouteName = state.routes[state.index].name;
+    trace.current = await perf().startTrace(`${currentRouteName}_LoadTime`);
+    trace.current.putAttribute("screenName", currentRouteName);
+    trace.current.putMetric("initTime", Date.now());
+  };
```

Comment:
We don't have to stop the execution using `await` for this, we can use `then()` to let the execution continue, without this affecting the program efficiency.

```javascript
  trace.current = perf().startTrace(`${currentRouteName}_LoadTime`);
  trace.current.then((traceInstance) => {
    traceInstance.putAttribute("screenName", currentRouteName);
    traceInstance.putMetric("initTime", Date.now());
    trace.current = traceInstance; 
  });
};
```

Reply:
@Gauravjeetsingh Veer ji Firebase, the recommendation is to use **async/ await**. Although using then() looks more efficient. However, I still think we should go with the documentation. Here is the [documentation](https://rnfirebase.io/perf/usage)

Reply:
@amitojsingh veerji, that's fine then. Keep an eye on performance, if it starts affecting performance, we might have to change.